Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -7676,40 +7676,123 @@ def to_display(self):
+    :param size: limit the number of returned results.
+
+    :type to_date: ``Optional[str]``
+    :param to_date: the end date to search until to.
+
+    :type value: ``str``
+    :param value: the indicator value to search.
+
+    :type limit ``Optional[int]``
+    :param limit the upper limit of the search
+
     :return: No data returned
     :rtype: ``None``
     """
-    def __init__(self, page=0, filter_fields=None):
+    def __init__(self,
+                 page=0,
+                 filter_fields=None,
+                 from_date=None,
+                 query=None,
+                 size=100,
+                 to_date=None,
+                 value='',
+                 limit=None):
         # searchAfter is available in searchIndicators from version 6.1.0
         self._can_use_search_after = is_demisto_version_ge('6.1.0')
         # populateFields merged in https://github.com/demisto/server/pull/18398
         self._can_use_filter_fields = is_demisto_version_ge('6.1.0', build_number='1095800')
         self._search_after_title = 'searchAfter'
         self._search_after_param = None
+        self._original_page = page
         self._page = page
         self._filter_fields = filter_fields
+        self._total = None
+        self._from_date = from_date
+        self._query = query
+        self._size = size
+        self._to_date = to_date
+        self._value = value
+        self._original_limit = limit
+        self._limit = limit
+
+    def __iter__(self):
+        self._total = None
+        self._search_after_param = None
+        self._page = self._original_page
+        self.limit = self._original_limit
+        return self
+
+    def __next__(self):
+        if self._is_search_done():
+            raise StopIteration
+        size = min(self._size, self.limit or self._size)
+        res = self.search_indicators_by_version(from_date=self._from_date,
+                                                query=self._query,
+                                                size=size,
+                                                to_date=self._to_date,
+                                                value=self._value)
+        fetched_len = len(res.get('iocs', []))
+        if fetched_len == 0:
+            raise StopIteration
+        if self.limit:
+            self.limit -= fetched_len
+        return res
+
+    @property
+    def page(self):
+        return self._page
+
+    @property
+    def total(self):
+        return self._total
+
+    @property
+    def limit(self):
+        return self._limit
+
+    @limit.setter
+    def limit(self, value):
+        self._limit = value
+
+    def _is_search_done(self):
+        """
+        Checks one of 2 conditions:
+        1. self._total was populated by a previous search, but no self._search_after_param
+        2. self.limit is set, and it's less or equal to zero
+        """
+        return (self._total and self._can_use_search_after and not self._search_after_param) or \
```

Comment:
if using paging: not _self._can_use_search_after:
check the options that it's the last page or that the number of results is smaller than page size